pkgs/applications/version-management/deepgit/default.nix

```diff
@@ -0,0 +1,89 @@
+{ fetchurl
+, glib
+, gnome
+, gtk3
+, jre
+, lib
+, makeDesktopItem
+, stdenv
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deepgit";
+  version = "4.3";
+
+  src = fetchurl {
+    url = "https://www.syntevo.com/downloads/${pname}/${pname}-linux-${builtins.replaceStrings [ "." ] [ "_" ] version}.tar.gz";
+    hash = "sha256-bA/EySZjuSDYaZplwHcpeP1VakcnG5K1hYTk7cSVbz0=";
+  };
+
+  nativeBuildInputs = [wrapGAppsHook];
+
+  buildInputs = [
+    gnome.adwaita-icon-theme
+    gtk3
+    jre
+  ];
+
+  preFixup = with lib; ''
+    gappsWrapperArgs+=( \
+      --prefix LD_LIBRARY_PATH : ${makeLibraryPath [
+        gtk3
+        glib
+      ]} \
+      --set DEEPGIT_JAVA_HOME ${jre} \
+    )
+    patchShebangs bin/${pname}.sh
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -pv $out/{bin,share/icons/hicolor/scalable/apps/}
+    cp -a lib license.html $out
+    cp bin/${pname}.sh $out/bin/${pname}
+
+    cp -av ${makeDesktopItem rec {
+      name = pname;
+      desktopName = "DeepGit";
+      keywords = [ "git" ];
+      comment = "Git-Client";
+      categories = [
+        "Development"
+        "RevisionControl"
+      ];
+      terminal = false;
+      startupNotify = true;
+      startupWMClass = desktopName;
+      exec = pname;
+      mimeTypes = [
+        "x-scheme-handler/${pname}"
+        "x-scheme-handler/sourcetree"
+      ];
+      icon = pname;
```

Comment:
Would be nice to un-inline this:
```nix
desktopItem = makeDesktopItem { /* ... */ };
installPhase = ''
  ln -s $desktopItem/share/applications $out/share
'';
```

Reply:
fair, just afraid to unnecessarily expose/mask something